### DISCUSSION TITLE
How to have a postUpgradeTasks command only ONCE per group / branch

### DISCUSSION THREAD

**Discussion by kenotron** [renovatebot/renovate#8993] [Request Help]
Right now, if I have a package rule that groups multiple deps to be placed in the same PR, it APPEARS that the postUpgradeTasks are run once per "upgrade" (which is every dependency).

The issue is, there is a need to only apply some post upgrade task right before publishing the PR, but only ONCE. I use this kind of flow for creating a manifest or a change file to describe a semver bump of internal package.json versions (the flow is defined in  https://rushjs.io/ or https://microsoft.github.io/beachball/).

Is this possible right now? Or is this a feature request?

**Comment by kenotron** [comment]
Ooops, I stand corrected, it's even **WORSE**. It calls the postUpgradeTask commands this many times:

- N packages in group
- M dependency upgrades
- C commands

N * M * C...

I am looking for some way to call a set of commands right before PR publishing.

**Comment by viceice** [comment]
Please follow #8220
